specs/_features/eip7594/p2p-interface.md

```diff
@@ -0,0 +1,180 @@
+      - [Samples subnets](#samples-subnets)
+        - [`data_column_sidecar_{subnet_id}`](#data_column_sidecar_subnet_id)
+  - [The Req/Resp domain](#the-reqresp-domain)
+    - [Messages](#messages)
+      - [DataColumnSidecarByRoot v1](#datacolumnsidecarbyroot-v1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Modifications in EIP-7594
+
+### Preset
+
+| Name | Value | Description |
+| - | - | - |
+| `KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH` | `uint64(floorlog2(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments')))` (= 4) | <!-- predefined --> Merkle proof index for `blob_kzg_commitments` |
+
+### Configuration
+
+| Name | Value | Description |
+| - | - | - |
+| `DATA_COLUMN_SIDECAR_SUBNET_COUNT` | `32` | The number of data column sidecar subnets used in the gossipsub protocol |
+
+### Containers
+
+#### `DataColumnSidecar`
+
+```python
+class DataColumnSidecar(Container):
+    index: ColumnIndex  # Index of column in extended matrix
+    column: DataColumn
+    kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    kzg_proofs: List[KZGProof, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    signed_block_header: SignedBeaconBlockHeader
+    kzg_commitments_inclusion_proof: Vector[Bytes32, KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH]
+```
+
+#### `DataColumnIdentifier`
+
+```python
+class DataColumnIdentifier(Container):
+    block_root: Root
+    index: ColumnIndex
+```
+
+### Helpers
+
+##### `verify_data_column_sidecar_kzg_proof`
+
+```python
+def verify_data_column_sidecar_kzg_proof(sidecar: DataColumnSidecar) -> bool:
+    """
+    Verify if the proofs are correct
+    """
+    assert sidecar.index < NUMBER_OF_COLUMNS
+    assert len(sidecar.column) == len(sidecar.kzg_commitments) == len(sidecar.kzg_proofs)
+    row_ids = [RowIndex(i) for i in range(len(sidecar.column))]
+
+    # KZG batch verifies that the cells match the corresponding commitments and proofs
+    return verify_cell_proof_batch(
+        row_commitments=sidecar.kzg_commitments,
+        row_indices=row_ids,  # all rows
+        column_indices=[sidecar.index],
+        cells=sidecar.column,
+        proofs=sidecar.kzg_proofs,
+    )
+```
+
+##### `verify_data_column_sidecar_inclusion_proof`
+
+```python
+def verify_data_column_sidecar_inclusion_proof(sidecar: DataColumnSidecar) -> bool:
+    """
+    Verify if the given KZG commitments included in the given beacon block.
+    """
+    gindex = get_subtree_index(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments'))
+    return is_valid_merkle_branch(
+        leaf=hash_tree_root(sidecar.kzg_commitments),
+        branch=sidecar.kzg_commitments_inclusion_proof,
+        depth=KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH,
+        index=gindex,
+        root=sidecar.signed_block_header.message.body_root,
+    )
+```
+
+##### `compute_subnet_for_data_column_sidecar`
+
+```python
+def compute_subnet_for_data_column_sidecar(column_index: ColumnIndex) -> SubnetID:
```

Comment:
I think this should be named `compute_subnet_for_column` instead because the sidecar is not the input.

Reply:
It is to be symmetric with `compute_subnet_for_blob_sidecar(blob_index: BlobIndex) -> SubnetID` helper (Deneb).